packages/stream-caip10-link-handler/src/caip10-link-handler.ts

```diff
@@ -110,11 +110,26 @@ export class Caip10LinkHandler implements StreamHandler<Caip10Link> {
     }
 
     // TODO: handle CAIP-10 addresses in proof generation of 3id-blockchain-utils
-    const account = validProof.account || validProof.address
-    let [address, chainId] = account.split('@') // eslint-disable-line prefer-const
+    const account: string = validProof.account || validProof.address
 
-    const addressCaip10 = [address, chainId].join('@')
-    if (addressCaip10.toLowerCase() !== state.metadata.controllers[0].toLowerCase()) {
+    let legacyAccountCaip10 = account
```

Comment:
why are we converting new-style caip10 addresses into the legacy format, instead of the other way around?

EDIT: I see, I guess it's because the controller will always be in legacy format to preserve backwards compatibility.

Reply:
yes, that's right.